presto-main/src/main/java/com/facebook/presto/resourcemanager/DistributedTaskInfoResource.java

```diff
@@ -0,0 +1,86 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.resourcemanager;
+
+import com.facebook.airlift.http.client.HttpUriBuilder;
+import com.facebook.presto.execution.TaskId;
+import com.facebook.presto.server.BasicQueryInfo;
+import com.facebook.presto.spi.QueryId;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+import java.util.Optional;
+
+import static com.facebook.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static com.facebook.presto.server.security.RoleType.ADMIN;
+import static com.facebook.presto.server.security.RoleType.USER;
+import static java.util.Objects.requireNonNull;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+
+@Path("/v1/taskInfo")
+@RolesAllowed({USER, ADMIN})
+public class DistributedTaskInfoResource
+{
+    private final ResourceManagerClusterStateProvider clusterStateProvider;
+    private final ResourceManagerProxy proxyHelper;
+
+    @Inject
+    public DistributedTaskInfoResource(
+            ResourceManagerClusterStateProvider clusterStateProvider,
+            ResourceManagerProxy proxyHelper)
+    {
+        this.clusterStateProvider = requireNonNull(clusterStateProvider, "clusterStateProvider is null");
+        this.proxyHelper = requireNonNull(proxyHelper, "proxyHelper is null");
+    }
+
+    @GET
+    @Path("{taskId}")
+    public void getTaskInfo(@PathParam("taskId") TaskId taskId,
+            @Context UriInfo uriInfo,
+            @Context HttpServletRequest servletRequest,
+            @Suspended AsyncResponse asyncResponse)
+            throws WebApplicationException
+    {
+        proxyTaskInfoResponse(servletRequest, asyncResponse, uriInfo, taskId);
+    }
+
+    private void proxyTaskInfoResponse(HttpServletRequest servletRequest, AsyncResponse asyncResponse, UriInfo uriInfo, TaskId taskId)
+    {
+        QueryId queryId = taskId.getQueryId();
+        Optional<BasicQueryInfo> queryInfo = clusterStateProvider.getClusterQueries().stream()
+                .filter(query -> query.getQueryId().equals(queryId))
+                .findFirst();
+
+        if (!queryInfo.isPresent()) {
+            asyncResponse.resume(Response.status(NOT_FOUND).type(MediaType.APPLICATION_JSON_TYPE).build());
+            return;
+        }
+
+        HttpUriBuilder uriBuilder = uriBuilderFrom(queryInfo.get().getSelf()).replacePath(uriInfo.getPath());
+        uriInfo.getQueryParameters().forEach(uriBuilder::addParameter);
+        proxyHelper.performRequest(servletRequest, asyncResponse, uriBuilder.build());
```

Comment:
let's keep this similar way as we have done it in `DistributedQueryInfoResource`?

Reply:
Here it's needed. The reason I added those two lines is that query parameters (like `includeLocalQueryOnly`) weren't getting preserved when proxying, so I copied them as well. Above I was copying that fix as well so that the code was consistent.

Reply:
Rolled it back